lightning_pose/data/datasets.py

```diff
@@ -335,3 +340,175 @@ def __getitem__(self, idx: int) -> HeatmapLabeledExampleDict:
+                                        {keypoint} != {keypointComp}")
+
+        self.data_length = list(self.data_length.values())[0]
+        for idx in range(self.data_length):
+            img_name_buff = []
+            for view, heatmaps in self.dataset.items():
+                img_name_buff.append(heatmaps.image_names[idx].split(delimiter)[-1])
+                if len(set(img_name_buff)) != 1:
+                    raise ImportError(f"Discrepancy in images names across CSV \
+                                      files! index:{idx}, image frame names:{img_name_buff}")
+
+    @property
+    def height(self) -> int:
+        return self.imgaug_transform[-1].get_parameters()[0][0].value
+
+    @property
+    def width(self) -> int:
+        # assume resizing transformation is the last imgaug one
+        return self.imgaug_transform[-1].get_parameters()[0][1].value
+
+    def __len__(self) -> int:
+        return self.data_length
+
+    @property
+    def output_shape(self) -> tuple:
+        return (
+            self.height // 2**self.downsample_factor,
+            self.width // 2**self.downsample_factor,
+        )
+
+    @property
+    def num_views(self) -> int:
+        return len(self.view_names)
+
+    def fusion(self, datadict: dict, bbox: Optional[np.array] = np.array((0))):
+        """ Here all the view data will be merged into one image and multiple heatmaps.
+            images and heatmaps will be concatenated vertically dim=1
+        Args:
+            heatmaps: this comes from HeatmapDataset.__getItems__(idx) for each view.
+            bbx: the part of the image that needs to be cropped
+        """
+        heatmaps = []
+        keypoints = []
+        images = []
+        concat_order = []
+        for view, data in datadict.items():
+            heatmaps.append(data["heatmaps"])
+            data["keypoints"] = data["keypoints"].reshape(int(data["keypoints"].shape[0] / 2), 2)
+            keypoints.append(data["keypoints"])
+            images.append(data["images"].unsqueeze(0))
+            concat_order.append(view)
+
+        heatmaps = torch.cat(heatmaps, dim=0)
+        image = torch.cat(images, dim=0)
+        keypoints = torch.cat(keypoints, dim=0).reshape(-1)
+
+        assert keypoints.shape == (self.num_targets,)
+
+        return image, heatmaps, keypoints, concat_order
+
+    def __getitem__(self, idx: int) -> MultiviewHeatmapLabeledBatchDict:
+        """Get an example from the dataset.
+        Calls the heatmapdataset for each csv file to get
+        Images and their heatmaps and then stacks them.
+        """
+        datadict = {}
+        img_sizes = {}
+        for view in self.view_names:
+            datadict[view] = self.dataset[view][idx]
+            img_sizes[view] = self.dataset[view].image_original_size[idx]
+        images, heatmaps, keypoints, concat_order = self.fusion(datadict)
+
+        return MultiviewHeatmapLabeledBatchDict(
```

Comment:
Batch -> Example (see below)

Reply:
should i also change dataset.py? line 501?